server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/EventDeadLettersRoutes.java

```diff
@@ -73,10 +81,42 @@ public String getBasePath() {
 
     @Override
     public void define(Service service) {
+        service.post(BASE_PATH, this::performActionOnAllEvents, jsonTransformer);
         service.get(BASE_PATH + "/groups", this::listGroups, jsonTransformer);
         service.get(BASE_PATH + "/groups/" + GROUP_PARAM, this::listFailedEvents, jsonTransformer);
+        service.post(BASE_PATH + "/groups/" + GROUP_PARAM, this::performActionOnGroupEvents, jsonTransformer);
         service.get(BASE_PATH + "/groups/" + GROUP_PARAM + "/" + EVENT_ID_PARAM, this::getEventDetails);
         service.delete(BASE_PATH + "/groups/" + GROUP_PARAM + "/" + EVENT_ID_PARAM, this::deleteEvent);
+        service.post(BASE_PATH + "/groups/" + GROUP_PARAM + "/" + EVENT_ID_PARAM, this::performActionOnSingleEvent, jsonTransformer);
+    }
+
+    @POST
+    @Path("")
+    @ApiOperation(value = "Performing operations on all events")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            dataType = "String",
+            name = "action",
+            paramType = "query",
+            example = "?action=reDeliver",
+            value = "Specify the action to perform on all events. For now only 'reDeliver' is supported as an action, "
+                + "and its purpose is to attempt a redelivery of all events present in dead letter."),
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "The taskId of the given scheduled task", response = TaskIdDto.class,
+            responseHeaders = {
+                @ResponseHeader(name = "Location", description = "URL of the resource associated with the scheduled task")
+            }),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = INVALID_ACTION_ARGUMENT_REQUEST),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = INTERNAL_SERVER_ERROR)
+    })
+    public TaskIdDto performActionOnAllEvents(Request request, Response response) {
```

Comment:
does it make sense to allow several reDeliver tasks on a given cluster?

Reply:
Well, I guess that is the same kind of discussion than the one started with Duc PR

I would answer that:
 - This concern is waaay more general than this PR and ongoing work
 - Task management in a cluster is clearly out of scope for this work
 - The current TaskManager is based on a single threaded executor, ensuring tasks runs one after one. Hence, in a single node context (the only one we can decently handle today), these 2 reDeliver operations will run one after each other. This let time for the admin to cancel them before it is too late.

Is the current behaviour satisfying?

Or do we want some kind of validation when submitting the tasks?

(Maybe we need to educate our admin, and teach them REST queries can schedule expensive things...)

Reply:
As discussed by chat, we agreed to implement a basic "single consumer" task manager to ensure serial execution of tasks on a cluster in a later sprint.
But we don't want to teach admins how not to break our brittle product, we want to make a admin-friendly and rock solid product!